processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -232,7 +240,16 @@ func setHostCPUInfo(d *Detector, cpuInfo cpu.InfoStat) error {
 	}
 
 	d.rb.SetHostCPUModelName(cpuInfo.ModelName)
-	d.rb.SetHostCPUStepping(int64(cpuInfo.Stepping))
+	if hostCPUSteppingAsStringFeatureGate.IsEnabled() {
+		// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31136
+		d.logger.Info("This attribute will change from int to string. Switch now using the feature gate.",
+			zap.String("attribute", "host.cpu.stepping"),
+			zap.String("feature gate", hostCPUSteppingAsStringID),
+		)
```

Comment:
I think this should go on the other if branch, right?

Reply:
Right, after the change to address https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31165#discussion_r1485952506 that one needs to be switched as well. Thank's for spotting this.